addon/components/uni-datepicker.js

```diff
@@ -16,16 +21,33 @@ export default Component.extend({
   dateFormat: 'MMMM YYYY',
   showDaysAround: true,
   disabledDates: [],
+  messages: [],
 
   isFirstMonth: computed('center', function() {
     return this.get('center').isSame(this.get('minDate'), 'month') ? 'disabled' : '';
   }),
   isLastMonth: computed('center', function() {
     return this.get('center').isSame(this.get('maxDate'), 'month') ? 'disabled' : '';
   }),
+  selectedMessage: computed('messages', function() {
+    let messagesArray = this.get('messages');
+
+    if (isEmpty(messagesArray)) {
+      return null;
+    }
+
+    return messagesArray.sort(function(a, b) {
+      if (a.type === b.type) {
```

Comment:
Why do you have a guard clauses mixed with ternary operators? Please use only guard clauses:

```js
if (a.type === b.type) {
  return 0;
}

if (a.type === MessageType.INFORMATION) {
  return 1;
}

return -1;
```